Sources/ProjectDescription/TargetAction.swift

```diff
@@ -10,20 +10,62 @@ public struct TargetAction: Codable, Equatable {
         case post
     }
 
+    /// How to execute the target action
+    ///
+    /// - file: Executes the tool, calling the script at the path. Tuist will look up the tool on the environment's PATH
+    /// - text: Executes the entered script. This should be a short command.
+    private enum Script: Equatable {
+        case file(_ tool: String?, _ path: Path?, _ args: [String])
+        case text(String)
+    }
+
     /// Name of the build phase when the project gets generated.
     public let name: String
 
+    /// The script to execute in the action
+    private let script: Script
+
+    public var isEmbeddedScript: Bool {
+        switch self.script {
+        case .file:
+            return false
+        case .text:
+            return true
+        }
+    }
+
     /// Name of the tool to execute. Tuist will look up the tool on the environment's PATH.
-    public let tool: String?
+    public var tool: String? {
+        switch script {
+        case .file(let tool, _, _):
```

Comment:
I think it's preferred to extract `let` before unwrapping the case - it's then easier to use the rest of the properties

Reply:
I'm not exactly sure what this means, but I've refactored to be an `if case` statement which should be easier to read & maintain.